RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -2895,22 +2903,27 @@ void PFAlgo::processBlock( const reco::PFBlockRef& blockref,
 				reco::PFBlock::LINKTEST_ALL );
 
       // Create a photon
-      auto& cand = (*pfCandidates_)[reconstructCluster( *eclusterref, sqrt(ecalSatellite.second.second.Mag2()) )];
-      cand.setEcalEnergy( eclusterref->energy(),sqrt(ecalSatellite.second.second.Mag2()) );
+      double ecalClusterEnergyCalibrated = sqrt(std::get<1>(ecalSatellite.second).Mag2()) * std::get<2>(ecalSatellite.second); // KH: calibrated under the egamma hypothesis (rawEcalClusterEnergy * calibration)
+      auto& cand = (*pfCandidates_)[reconstructCluster( *eclusterref, ecalClusterEnergyCalibrated )];
+      cand.setEcalEnergy( eclusterref->energy(), ecalClusterEnergyCalibrated );
       cand.setHcalEnergy( 0., 0. );
       cand.setHoEnergy( 0., 0. );
       cand.setPs1Energy( associatedPSs[iEcal].first );
       cand.setPs2Energy( associatedPSs[iEcal].second );
       cand.addElementInBlock( blockref, iEcal );
       cand.addElementInBlock( blockref, sortedTracks.begin()->second) ;
-    }
-
-
-  }
 
-   // end loop on hcal element iHcal= hcalIs[i]
+      if ( fabs(eclusterref->energy() - sqrt(std::get<1>(ecalSatellite.second).Mag2()))>1e-3 || fabs(eclusterref->correctedEnergy() - ecalClusterEnergyCalibrated)>1e-3 )
+	edm::LogWarning("PFAlgo:processBlock") << "ecalCluster vs ecalSatellites look inconsistent (eCluster E, calibE, ecalSatellite E, calib E): "
+					       << eclusterref->energy() << " " << eclusterref->correctedEnergy() << " "
```

Comment:
What are the implications on the ecalSatellites data (or what follows from them)?
If the data is usable, this warning better be replaced by a LogInfo or put behind ```if (debug_```.
If the data is completely unusable and/or if this simply is not supposed to happen (e.g. indicates a bug in the code), perhaps even an exception is a better solution.
@hatakeyamak  please clarify

In my tests I did not see any instance of this happening out of about 7K events with 2018-like pileup.
So, at least from the point of view of frequency, this is OK to stay as a warning (as it does no harm so far).